### PR TITLE
LPD-39218 Validate allowed DL File Extensions in MB

### PR DESCRIPTION
**Pull request by robertoDiaz**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-39218

Avoid let users add any extension type as MB attachment.

# How am I fixing it?

Adding a validation to limit the allowed MB attachments extensions based in DL limitations

# How can you verify that it works?

Run the included automated tests.

### CODE REVIEW

File: modules/apps/message-boards/message-boards-web/src/main/java/com/liferay/message/boards/web/internal/upload/BaseMBUploadFileEntryHandler.java [hunk 1]
```diff
@@ -41,10 +41,14 @@ public FileEntry upload(UploadPortletRequest uploadPortletRequest)
 			(ThemeDisplay)uploadPortletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
+		String fileName = _getFileName(uploadPortletRequest);
+
+		_dlValidator.validateFileExtension(fileName);
+
+		String contentType = _getContentType(uploadPortletRequest);
```
Comment (AliciaGarciaGarcia) [inline modules/apps/message-boards/message-boards-web/src/main/java/com/liferay/message/boards/web/internal/upload/BaseMBUploadFileEntryHandler.java:48]: maybe this is no need to be extracted, but ok anyway

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by jkappler** [comment]
unit test failure is a known issue, so I think it's good to forward @AliciaGarciaGarcia and @robertoDiaz :heart:

**Comment by jkappler** [comment]
ci:forward:force

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**